### PR TITLE
fix: Rework workspace visibility flow

### LINKED ISSUE #2777: Bug: Rework workspace visibility flow in the workspace settings page
The **Change Workspace Visibility** section should be about the **Upgrade your workspace**

The Set to private button should be a primary CTA button and enabled. When clicked, if the user has not upgraded, they get the upgrade workspace modal flow, otherwise, the workspace gets set to private.

CleanShot 2024-02-26 at 17 00 56

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Moves the visibility setting above the upgrade card in the workspace settings. It will show the upgrade modal if the user is not in a PRO workspace.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2777 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/20603494/e1e0cfc6-b7ef-4cea-8ed6-cbf83db9dc72

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by jpmcb — APPROVED** [review]
QA'd - looks solid!

File: components/Workspaces/InsightUpgradeModal.tsx [hunk 1]
```diff
@@ -26,19 +26,30 @@ export default function InsightUpgradeModal({
...
-                  {overLimit} {variant}
-                </span>{" "}
-                but the free Workspace only allows for{" "}
-                <span className="font-bold">
-                  {variant === "repositories" ? 100 : 10} {variant}
-                </span>{" "}
-                tracked. Don&apos;t worry, your insights won&apos;t be deleted, but if you want to continue using
-                OpenSauced you should upgrade your Workspace to a PRO Workspace.
-              </p>
+              <DialogTitle className="text-xl">
```
Comment (nickytonline) [inline components/Workspaces/InsightUpgradeModal.tsx:29]: We'll need to rework this to support the newer dialog/drawer component @isabensusan, but not a blocker atm.

**Review by nickytonline — APPROVED** [review]
:shipit:

**Comment by zeucapua** [comment]
Before merging, I'd love some copy for this modal! cc @BekahHW @isabensusan

File: components/Workspaces/InsightUpgradeModal.tsx [hunk 2]
```diff
@@ -26,19 +26,30 @@ export default function InsightUpgradeModal({
...
+                  Your Insight page has{" "}
+                  <span className="font-bold">
+                    {overLimit} {variant}
+                  </span>{" "}
+                  but the free Workspace only allows for{" "}
+                  <span className="font-bold">
+                    {variant === "repositories" ? 100 : 10} {variant}
+                  </span>{" "}
+                  tracked. Don&apos;t worry, your insights won&apos;t be deleted, but if you want to continue using
+                  OpenSauced you should upgrade your Workspace to a PRO Workspace.
```
Comment (BekahHW) [inline components/Workspaces/InsightUpgradeModal.tsx:45]: Don't worry, your insights won't be deleted. If you want to continue adding to this Insight Page, you'll need to upgrade your Workspace to a PRO Account.

**Comment by isabensusan** [comment]
@zeucapua very small comment -- I know I wrote the copy 🙈  but seeing it live: can we match the wording and make it ".. is _over_ the limit" instead of _past_ the limit in both?

image

image

**Comment by zeucapua** [comment]
I can change the copy! As for the full width, I don't think I can since the WorkspaceLayout gives the padding and I don't want to mess with it lest I break something @nickytonline fixed for the UI. I can try, but it's not guaranteed!

**Comment by zeucapua** [comment]
I'll do the banner changes in #2809

**Pull request merged**